beacon/headers/header-referrer.js

```diff
@@ -3,15 +3,18 @@ var RESOURCES_DIR = "/beacon/resources/";
 var referrerOrigin = self.location.origin + '/';
 var referrerUrl = self.location.href;
 
-function testReferrerHeader(testBase, expectedReferrer) {
+function testReferrerHeader(testBase, expectedReferrer, mayBeBlockedAsMixedContent = false) {
   var id = self.token();
   var testUrl = testBase + "inspect-header.py?header=referer&cmd=put&id=" + id;
 
   promise_test(function(test) {
-    assert_true(navigator.sendBeacon(testUrl), "SendBeacon Succeeded");
+    const sentBeacon = navigator.sendBeacon(testUrl);
+    assert_true(sentBeacon || mayBeBlockedAsMixedContent, "SendBeacon Succeeded");
+    if (!sentBeacon && mayBeBlockedAsMixedContent)
```

Comment:
I think this would be easier to read if the pass/fail logic wasn't separated into the assert and the if (!sentBeacon)